stripe/src/main/java/com/stripe/android/model/CardBrand.kt

```diff
@@ -279,10 +297,28 @@ enum class CardBrand(
                 return Unknown
             }
 
-            return values()
-                .firstOrNull { cardBrand ->
+            // Only return a card brand if we know exactly which one, if there is more than
+            // one possibility return unknown
+            return (
+                values().filter { cardBrand ->
```

Comment:
Same logic used here and in `getCardBrands`

Might be nice to extract this to a separate method:

```
values().filter { cardBrand ->
    cardBrand.getPatternForLength(cardNumber)?.matcher(cardNumber)
        ?.matches() == true
}
```